sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Azure.Messaging.EventHubs.Processor.Tests.csproj

```diff
@@ -5,6 +5,7 @@
     <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
     <ExcludeRecordingFramework>true</ExcludeRecordingFramework>
+    <DefineConstants>$(DefineConstants);EXCLUDE_RECORDING</DefineConstants>
```

Comment:
Shouldn't excluding the recording framework imply that recording is not included?    

Maybe the `TestFramework.props` could define a `RECORDING_FRAMEWORK_EXCLUDED` constant automatically when the framework is not included so that each service project doesn't need to opt out twice?

Reply:
I'm also wondering if knowledge of the recording framework should be built into the client test fixture or if there is a way to separate them.  Maybe we could consider doing a partial class that gets included/excluded with the recording framework instead of using a compiler constant?

Reply:
I like the idea of partial class. @pakrym , what do you think?

Reply:
I wonder if we should make this feature less magical. Like having a parameter on InstrumentClientOptions or something similar, that would allow us to be more targeted and flexible with what clients get simulated failures and also get rid of this problem with shared source.

Not having retry simulation be tied to NUnit would also be nice if we don't have to.